### PR TITLE
Allow the setting of log level by env vars

### PR DESCRIPTION
**Pull request by adamdougal**
Required by https://github.com/Azure-Samples/chat-with-your-data-solution-accelerator/issues/637

### CODE REVIEW

**Review by ross-p-smith — APPROVED** [review]

File: code/create_app.py [hunk 1]
```diff
@@ -295,6 +295,8 @@ def create_app():
     app = Flask(__name__)
     env_helper: EnvHelper = EnvHelper()
 
+    logging.basicConfig(level=env_helper.LOGLEVEL)
```
[the remark this answers is not in the document]
Comment (adamdougal) [reply code/create_app.py:298]: I will double check 👍

**Review by ross-p-smith — DISMISSED** [review]

**Comment by adamdougal** [comment]
Just doing some further testing and making it easy to configure via Bicep

**Comment by adamdougal** [comment]
Changes since last reviewed:
- Made it easy to configure log level via bicep
- Ensured logs are appearing in app insights correctly
- This involved setting the level individually in each function file

**Comment by adamdougal** [comment]
> We seem to be now getting all the logs from the azure packages which are really verbose, could we set the log level to WARN?

I'm wondering if it might actually be useful to have these logs. Do you think it's _too_ verbose?

> Out of interest, did you try doing `logging.getLogger().setLevel(env_helper.LOGLEVEL)` in `function_app.py`?

I did but it had no affect unfortunately :(

**Comment by cecheta** [comment]
> > We seem to be now getting all the logs from the azure packages which are really verbose, could we set the log level to WARN?
> 
> I'm wondering if it might actually be useful to have these logs. Do you think it's _too_ verbose?

It's **super** verbose, it's practically impossible to see the actual logs coming from our application, especially when running locally, because the get lost in a sea of request/response logs.

**Comment by adamdougal** [comment]
> > > We seem to be now getting all the logs from the azure packages which are really verbose, could we set the log level to WARN?
> > 
> > 
> > I'm wondering if it might actually be useful to have these logs. Do you think it's _too_ verbose?
> 
> It's **super** verbose, it's practically impossible to see the actual logs coming from our application, especially when running locally, because the get lost in a sea of request/response logs.

Fair 😅 Let me downgrade it!

**Review by cecheta — APPROVED** [review]
Looks great, thank you so much!

**Pull request merged**